src/System Application/App/Json/src/JsonImpl.Codeunit.al

```diff
@@ -0,0 +1,417 @@
+                    end;
+                end;
+            FieldType::BLOB:
+                if TryReadAsBase64(FieldRef, Value) then
+                    Success := true;
+            FieldType::RecordID:
+                begin
+                    Success := Evaluate(RecID, Value);
+                    FieldRef.Value(RecID);
+                end;
+        end;
+
+        exit(Success);
+    end;
+
+    [TryFunction]
+    local procedure TryReadAsBase64(var BlobFieldRef: FieldRef; Value: Text)
+    var
+        Base64Convert: Codeunit "Base64 Convert";
+        TempBlob: Codeunit "Temp Blob";
+        RecordRef: RecordRef;
+        OutStream: OutStream;
+    begin
+        TempBlob.CreateOutStream(OutStream);
+        Base64Convert.FromBase64(Value, OutStream);
+        RecordRef := BlobFieldRef.Record();
+        TempBlob.ToRecordRef(RecordRef, BlobFieldRef.Number);
+    end;
+
+    local procedure TextToOptionValue(InputText: Text; OptionString: Text): Integer
+    var
+        IntVar: Integer;
+        Counter: Integer;
+    begin
+        if InputText = '' then
+            InputText := ' ';
+
+        if Evaluate(IntVar, InputText) then begin
+            if IntVar < 0 then
+                IntVar := -1;
+            if GetOptionsQuantity(OptionString) < IntVar then
+                IntVar := -1;
+        end else begin
+            IntVar := -1;
+            for Counter := 1 to GetOptionsQuantity(OptionString) + 1 do
+                if UpperCase(GetSubStrByNo(Counter, OptionString)) = UpperCase(InputText) then
+                    IntVar := Counter - 1;
+        end;
+
+        exit(IntVar);
+    end;
+
+    local procedure GetOptionsQuantity(OptionString: Text): Integer
+    var
+        Counter: Integer;
+        CommaPosition: Integer;
+    begin
+        if StrPos(OptionString, ',') = 0 then
+            exit(0);
+
+        repeat
+            CommaPosition := StrPos(OptionString, ',');
+            OptionString := DelStr(OptionString, 1, CommaPosition);
+            Counter := Counter + 1;
+        until CommaPosition = 0;
+
+        exit(Counter - 1);
+    end;
+
+    local procedure GetSubStrByNo(Number: Integer; CommaString: Text) SelectedStr: Text
+    var
+        SubStrQuantity: Integer;
+        Counter: Integer;
+        CommaPosition: Integer;
+    begin
+        if Number <= 0 then
+            exit;
+
+        SubStrQuantity := GetOptionsQuantity(CommaString);
+        if SubStrQuantity + 1 < Number then
+            exit;
+
+        repeat
+            Counter := Counter + 1;
+            CommaPosition := StrPos(CommaString, ',');
+            if CommaPosition = 0 then
+                SelectedStr := CommaString
+            else begin
+                SelectedStr := CopyStr(CommaString, 1, CommaPosition - 1);
+                CommaString := DelStr(CommaString, 1, CommaPosition);
+            end;
+        until Counter = Number;
+    end;
+
+    local procedure AddJPropertyToJObject(propertyName: Text; value: Variant)
```

Comment:
The casing in some of the parameters is triggering my OCD - but maybe you have a good reason for it? ;-)

Reply:
Ah, it's like that in the BaseApp version too...

Reply:
And also GitHub copilot decision, blame it ;)